R/generate_cre_dataset.R

```diff
@@ -11,79 +11,102 @@
 #' @param n_rules The number of causal rules, either 2 (default) or 4.
 #' @param effect_size The effect size magnitude (default: 2).
 #' @param p The number of covariates (default: 10).
+#' @param binary_covariates Whether to use binary or continuous covariates
+#' (default: TRUE)
 #' @param binary_outcome Whether to use binary or continuous outcomes
 #' (default: TRUE)
+#' @param confounding Only for continuous outcome,add confounding variables:
+#' linear confounding "lc", non-linear confounding "nlc", no confounding "nc"
+#' (default: "nc")
 #'
 #' @return
-#' A list of synthetic data containing an outcome vector (y), a treatment
-#' vector (z), and a matrix of p covariates (X).
+#' A list of synthetic data containing:
+#' - the outcome vector (y),
+#' - the treatment vector (z),
+#' - the covariates matrix (X) and
+#' - the individual treatment vector (ite)
+#'
+#' @note
```

Comment:
The note may need some improvement, it adds more confusion than clarity. What is the range of possible effect size? How decreasing the absolute value of the causal effect increase the complexity of the synthetic data set? How do you define the complexity of the synthetic data set?

Reply:
it's complexity for heterogeneity discovery, not on the data itself
fixed